code/modules/crafting/glassware.dm

```diff
@@ -0,0 +1,299 @@
+//This file is for glass working types of things!
+
+/obj/item/glasswork
+	name = "This is a bug report it!"
+	desc = "Failer to code. Contact your local bug remover..."
+	icon = 'icons/obj/glassworks.dmi'
+	w_class = WEIGHT_CLASS_SMALL
+	force = 1
+	throw_speed = 1
+	throw_range = 3
+	tool_behaviour = null
+
+/obj/item/glasswork/glasskit
+	name = "Glass Working Tools"
+	desc = "A lovely belt of most the tools you will need to shape, mold, and refine glass into more advanced shapes."
+	icon_state = "glass_tools"
+	tool_behaviour = TOOL_GLASS_CUT
+
+/obj/item/glasswork/blowing_rod
+	name = "Glass Working Blow Rod"
+	desc = "A hollow metal stick made for glass blowing."
+	icon_state = "blowing_rods_unused"
+	tool_behaviour = TOOL_BLOW
+
+/obj/item/glasswork/glass_base
+	name = "Glass Fodder Sheet"
+	desc = "A sheet of glass set aside for glass working"
+	icon_state = "glass_base"
+	var/next_step = null
+	var/rod = /obj/item/glasswork/blowing_rod
+
+/obj/item/lens
+	name = "Optical Lens"
+	desc = "Good for selling or crafting, by itself its useless"
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "glass_optics"
+
+//////////////////////Chem Disk/////////////////////
+//Two Steps                                       //
+//Sells for 300 cr, takes 10 glass shets          //
+//Usefull for chem spliting                       //
+////////////////////////////////////////////////////
+
+/obj/item/glasswork/glass_base/dish
+	name = "Glass Fodder Sheet"
+	desc = "A set of glass sheets set aside for glass working, this one is ideal for a small glass dish. Needs to be cut with some tools."
+	next_step = /obj/item/glasswork/glass_base/dish_part1
+
+/obj/item/glasswork/glass_base/dish/attackby(obj/item/I, mob/user, params)
+	..()
+	if(I.tool_behaviour == TOOL_GLASS_CUT)
+		new next_step(user.loc, 1)
+		qdel(src)
```

Comment:
I see lot of this copypaste around, could you please move it to their parent types (and perhaps add an extra check to avoid bad things from happening should next_step be null) instead?
Heck, maybe make a new proc that can be overriden for certain peculiar steps such as the glass fodder sheet which also deletes the tool.

Reply:
Funnily enough, I suggested this because of the whole "parent type proc is getting called before subtype proc" thing.